wordpress-mobile/WordPress-Android
# [Jetpack Social] Migrate external services endpoint to v2

Fixes #18430 

To test:
1 - Download an older version of JP and install;
2 - Log in and select a site that has at least one social connection;
3 - Navigate to `Sharing` screen;
4 - Take note of the services and connections that are shown;
5 - Install the version from this branch;
6 - Navigate to `Sharing` screen;
7 - The services and connections that are shown should be the same as step 4;
8 - Try disconnecting and connecting again to a social media service and see if the status is correct;
9 - Try sharing a post to a connected social media service;
10 - Repeat the same process with WP;
11 - Repeat the same process with a fresh install (instead of version migration).

## Regression Notes
1. Potential unintended areas of impact
`Sharing` screen in `My Site` menu 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests

3. What automated tests I added (or what prevented me from doing so)
Implemented `PublicizeListViewModelTest`

PR submission checklist:


UI Changes testing checklist:
